src/framework/gracefulShutdown.ts

```diff
@@ -1,12 +1,35 @@
+import { logger } from "./logger.js";
 import { Application } from "../shared/types.js";
 import * as mongo from "./mongo.js";
+import { createTerminus, TerminusOptions } from "@godaddy/terminus";
+import { isDBHealthy } from "./crud-db.js";
 
-export const gracefulShutdown = (server: Application) => {
-  const atExitHandler = () => {
-    server.close();
-    mongo.disconnectDB();
-  };
+const loggerWrapper = (msg: string, err: Error) => {
+  if (err) {
+    logger.error(err.message);
+  } else {
+    logger.info(msg);
+  }
+};
 
-  process.once("SIGINT", atExitHandler);
-  process.once("SIGTERM", atExitHandler);
+export const gracefulShutdown = (app: Application) => {
+  const terminusOptions: TerminusOptions = {
+    onShutdown: async () => {
+      logger.info("Server shutting down");
+    },
+    onSignal: async () => {
+      await Promise.all([app.close(), mongo.disconnectDB()]);
```

Comment:
you should actually wait for the server to close before disconnecting from the DB to not stop ongoing requests